test/e2e/egress_test.go

```diff
@@ -664,6 +666,175 @@ func testEgressMigration(t *testing.T, data *TestData, triggerFunc, revertFunc f
+			originalRateLimit: rateLimit1,
+			newRateLimit:      nil,
+		},
+		{
+			name:              "change rate-limit",
+			originalRateLimit: rateLimit1,
+			newRateLimit:      rateLimit2,
+		},
+	}
+	checkRateLimitInstall := func(rateLimit *v1beta1.RateLimit) {
+		meterIDOnLocal := data.getEgressMeterID(egressNodeAntreaPodName, rateLimit.Peak, rateLimit.Burst)
+		if meterIDOnLocal == 0 {
+			t.Fatalf("Meter installation failed")
+		}
+		meterIDOnRemote := data.getEgressMeterID(nonEgressNodeAntreaPodName, rateLimit.Peak, rateLimit.Burst)
+		if meterIDOnRemote != 0 {
+			t.Fatalf("Unexpected meters present on non-EgressNode")
+		}
+		meterFlowNum := data.getEgressMeterFlowNum(egressNodeAntreaPodName, meterIDOnLocal)
+		// One local Pod flow and one tunnel flow.
+		if meterFlowNum != 2 {
+			t.Fatalf("Flows with meter installation failed")
+		}
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			egress := data.createEgress(t, "egress-qos-", matchExpressions, nil, "", egressNodeIP, tt.originalRateLimit)
+			_, err := data.waitForEgressRealized(egress)
+			if err != nil {
+				t.Fatalf("Error when waiting for Egress to be realized: %v", err)
+			}
+			if tt.originalRateLimit != nil {
+				checkRateLimitInstall(tt.originalRateLimit)
+			}
+
+			toUpdate := egress.DeepCopy()
+			err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				toUpdate.Spec.RateLimit = tt.newRateLimit
+				_, err := data.crdClient.CrdV1beta1().Egresses().Update(context.TODO(), toUpdate, metav1.UpdateOptions{})
+				if err != nil && errors.IsConflict(err) {
+					toUpdate, _ = data.crdClient.CrdV1beta1().Egresses().Get(context.TODO(), egress.Name, metav1.GetOptions{})
+				}
+				return err
+			})
+			require.NoError(t, err, "Failed to update Egress")
+
+			if tt.originalRateLimit != nil {
+				meterID := data.getEgressMeterID(egressNodeAntreaPodName, tt.originalRateLimit.Peak, tt.originalRateLimit.Burst)
+				if meterID != 0 {
+					t.Fatalf("Meter update/delete failed")
+				}
+
+				meterFlowNum := data.getEgressMeterFlowNum(egressNodeAntreaPodName, meterID)
+				if meterFlowNum != 0 {
+					t.Fatalf("Flows with meter update/delete failed")
+				}
+			}
+			if tt.newRateLimit != nil {
+				checkRateLimitInstall(tt.newRateLimit)
+			}
+			failOnError(data.deleteEgress(egress.Name), t)
+		})
+	}
+}
+
+func (data *TestData) getEgressMeterFlowNum(antreaPodName string, meterID int) int {
+	// dump-flows output should look like below:
+	// cookie=0x7040000000000, duration=17.079s, table=EgressMark, n_packets=0, n_bytes=0, idle_age=17, priority=200,ct_state=+new+trk,ip,tun_dst=172.18.0.3 actions=meter:1,set_field:0x1/0xff->pkt_mark,set_field:0x20/0xf0->reg0,goto_table:L2ForwardingCalc
+	// cookie=0x7040000000000, duration=17.073s, table=EgressMark, n_packets=0, n_bytes=0, idle_age=17, priority=200,ct_state=+new+trk,ip,in_port="se1-5584-d7ca50" actions=meter:1,set_field:0x1/0xff->pkt_mark,set_field:0x20/0xf0->reg0,goto_table:L2ForwardingCalc
```

Comment:
This is a bit implementation specific, and from verification's perspective, having the flow/meter installed doesn't always mean the rate limit is working as expected. Can we test actual bandwidth instead?